skactiveml/utils/_validation.py

```diff
@@ -664,11 +665,52 @@ def check_type(
         raise TypeError(error_str + ".")
 
 
+def check_callable(func, name, n_free_parameters=None):
+    """Checks if function is a callable and if the number of free parameters is
+    correct.
+
+    Parameters
+    ----------
+    func: callable
+        The functions to be validated.
+    name: str
+        The name of the function
+    n_free_parameters: int, optional (default=None)
+        The number of free parameters. If `n_free_parameters` is `None`,
+        `n_free_parameters` is set to `1`.
+    """
+
+    if n_free_parameters is None:
+        n_free_parameters = 1
+
+    if not callable(func):
+        raise TypeError(
+            f"`{name}` must be callable. " f"`{name}` is of type {type(func)}"
+        )
+
+    # count the number of arguments that have no default value
+    n_free_params = len(
```

Comment:
```suggestion
    n_actual_free_parameters = len(
```
What is the meaning of free parameters? Do you just mean the mandatory parameters without default values? Then maybe use `n_mandatory_parameters` and `n_actual_mandatory_parameters`.